src/sdk/endpoint_stats_collector.js

```diff
@@ -1,258 +1,289 @@
+    /**
+     * @param {NsfsStats} data
+     * @returns {Promise<void>}
+     */
+    async _process_nsfs_stats(data) {
+        dbg.log0('nsfs stats - IO counters :', data.io_stats);
+        for (const [k, v] of Object.entries(data.op_stats ?? {})) {
+            dbg.log0(`nsfs stats - S3 op=${k} :`, v);
+        }
+        for (const [k, v] of Object.entries(data.fs_workers_stats ?? {})) {
+            dbg.log0(`nsfs stats - FS op=${k} :`, v);
+        }
+
+        if (this.rpc_client) {
+            await this.rpc_client.stats.update_nsfs_stats({
+                nsfs_stats: data
+            }, {
                 timeout: SEND_STATS_TIMEOUT
             });
-            this.reset_all_nsfs_stats();
-        } catch (err) {
-            dbg.error('failed on update_nsfs_stats. trigger _send_nsfs_stats again', err);
-            this._trigger_send_nsfs_stats(attempts);
         }
     }
 
-    _new_namespace_stats() {
-        return {
-            read_count: 0,
-            write_count: 0,
-            read_bytes: 0,
-            write_bytes: 0,
-            error_write_bytes: 0,
-            error_write_count: 0,
-            error_read_bytes: 0,
-            error_read_count: 0,
-        };
-    }
-
-    _new_namespace_nsfs_stats() {
-        return {
-            read_count: 0,
-            write_count: 0,
-            read_bytes: 0,
-            write_bytes: 0,
-        };
+    _update_fs_stats(fs_worker_stats) {
+        const time = Math.floor(fs_worker_stats.took_time * 1000); // microsec
+        const op_name = fs_worker_stats.name.toLowerCase();
+        const error = fs_worker_stats.error;
+        this.nsfs_stats_collector.update({
+            fs_workers_stats: {
+                [op_name]: {
+                    count: 1,
+                    error_count: error,
+                    min_time: error ? undefined : time,
+                    max_time: error ? undefined : time,
+                    sum_time: error ? undefined : time,
+                }
+            }
+        });
     }
 
-    update_namespace_read_stats({ namespace_resource_id, bucket_name, size = 0, count = 0, is_err }) {
-        this.namespace_stats[namespace_resource_id] = this.namespace_stats[namespace_resource_id] || this._new_namespace_stats();
-        const io_stats = this.namespace_stats[namespace_resource_id];
-        if (is_err) {
-            io_stats.error_read_count += count;
-            io_stats.error_read_bytes += size;
-        } else {
-            io_stats.read_count += count;
-            io_stats.read_bytes += size;
-        }
+    update_namespace_read_stats({ namespace_resource_id, bucket_name = undefined, size = 0, count = 0, is_err = false }) {
+        this.endpoint_stats_collector.update({
+            namespace_stats: {
+                [namespace_resource_id]: is_err ? {
+                    error_read_count: count,
+                    error_read_bytes: size,
```

Comment:
I think we will never get is_err true

Reply:
yeah I see this is not used right now, but still I left the original option in case we want to add error stats.